content/module-3/1-create-your-messaging-protocol.md

```diff
@@ -0,0 +1,251 @@
+
+3- A specific transaction is sent to the receiving subnet in order to validate and execute the cross-subnet message
+
+4- An arbitary business logic is executed on the receiving subnet
+
+As mentioned before, all a messaging protocol developer needs to care about, are points 1. and 4., the rest being inherited for free by **ToposMessaging**.
+
+<HighlightBox type="info" title="Example: ERC20Messaging">
+
+**ERC20Messaging**, being a messaging protocol to send ERC20 tokens across subnets, implements the points 1. and 4. as follows:
+
+1- Burn the amount that the caller requested to send to another subnet
+
+4- Verify that the address of the **ERC20Messaging** contract on the sending subnet corresponds to the current one (i.e., on the receiving subnet), verify that the requested receiving subnet is the correct one, retrieve the requested ERC20 token, and finally mint the right amount to the right recipient
+
+</HighlightBox>
+
+Now, we will detail the ToposMessaging interface and its relevant methods for you to start leveraging its power.
+
+## ToposMessaging
+
+Like messaging protocol contracts inheriting it, **ToposMessaging** exposes functionalities that relate either to the sending subnet or the receiving subnet in a cross-subnet messaging flow.
+
+### Sending subnet (1)
+
+- `_emitMessageSentEvent(SubnetId targetSubnetId)`: This function expects a target subnet id which is the id of the receiving subnet, and takes care of emitting the specific event mentioned in Messaging protocol flow, point 2. This event is **ToposCore**'s **CrossSubnetMessageSent** event, which is not defined in **ToposMessaging**, hence the `_emitMessageSentEvent` helper function.
+
+### Receiving subnet (1)
+
+- `execute(uint256[] calldata logIndexes, bytes calldata proofBlob, bytes32 receiptRoot)`: This is the function to be called as part of the transaction mentioned in Messaging protocol flow, point 3.
+
+Let's dive deeper into its signature and implementation.
+
+As you can see, calling the `execute` function, i.e., executing a cross-subnet message, demands an array of log indexes, a proof blob, and a receipt root. 
+
+The `proof` in question is a merkle proof proving the inclusion of a receipt in a certain receipt trie (hence the receipt trie root third argument). That receipt is the one of the first transaction that we have listed above, the one executing on the sending subnet the arbitrary business logic at the origin of the cross-subnet message (Messaging protocol flow, point 1). This receipt plays a critical role in the cross-subnet message execution for it ensures the transaction on the sending subnet was executed correctly and it contains the logs which speak to the business logic of the messaging protocol.
+
+<HighlightBox type="info" title="Example: ERC20Messaging">
+
+**ERC20Messaging** emits, on a sending subnet, the following `TokenSent` event to detail which cross-subnet ERC20 transfer was requested:
+
+```
```

Comment:
```suggestion
```solidity
```

The syntax highlighter that we use handles Solidity.

Reply:
Nice, I didn't expect the syntax highlighter to work on the dev portal (thought this was github only).